Adding WaitForDownload test for .NET

halex2005 opened the pull request:
### **User description**
Hi!

### Description

I've added WaitForDownload test case for .NET.
Review, please.

Note, that I've also added `--no-sandbox` switch to ChromeOptions because chrome crashes without it.
It's to make sure that tests are green.

### Types of changes
- [ ] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [x ] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist
- [x ] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x ] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.


___

### **PR Type**
Tests, Enhancement


___

### **Description**
- Added a new `WaitForDownload` test method in `NetworkTest.cs` to handle and verify file downloads.
- Enhanced `BaseTest.cs` by adding a `--no-sandbox` argument to ChromeOptions to prevent Chrome crashes during tests.
- Updated documentation files to include CSharp code block references for better clarity and guidance.



___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>BaseTest.cs</strong><dd><code>Add no-sandbox argument to ChromeOptions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/dotnet/SeleniumDocs/BaseTest.cs

<li>Added <code>--no-sandbox</code> argument to ChromeOptions.<br> <li> Ensures Chrome does not crash during tests.<br>


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1885/files#diff-13e7f59b8933cf7834665df17e6282b912e7b6a94eb3d3ea51e02517365c77ce">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>NetworkTest.cs</strong><dd><code>Add WaitForDownload test method for file downloads</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/dotnet/SeleniumDocs/BiDi/CDP/NetworkTest.cs

<li>Added <code>WaitForDownload</code> test method.<br> <li> Implemented download behavior settings and event handling.<br> <li> Verified file download completion and existence.<br>


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1885/files#diff-e31523c285a75f1ed30aba52eb3c36546df30c19cd434a14ffbf02c431da54fc">+47/-3</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>network.en.md</strong><dd><code>Update CSharp tab with code block reference</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/bidi/cdp/network.en.md

- Updated CSharp tab with code block reference.



</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1885/files#diff-e2f70111bf20dada3969bcb94cd6e13d8b79a92875244b34885e5bf255d99e98">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>network.ja.md</strong><dd><code>Update CSharp tab with code block reference</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/bidi/cdp/network.ja.md

- Updated CSharp tab with code block reference.



</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1885/files#diff-71ba93565ff6c772351a15a2e9c923d2de92e5d5ab212d3d35940cab0b1bcc9f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>network.pt-br.md</strong><dd><code>Update CSharp tab with code block reference</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/bidi/cdp/network.pt-br.md

- Updated CSharp tab with code block reference.



</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1885/files#diff-271c98862bdd772a62a7e37fd2b18d44c41b7e8cecc80c264f60ba19a6d1edd7">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>network.zh-cn.md</strong><dd><code>Update CSharp tab with code block reference</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/bidi/cdp/network.zh-cn.md

- Updated CSharp tab with code block reference.



</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1885/files#diff-bd3f997b69e4212aee029c3e1046918d68e0bf499412735f9858feb278c9df55">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

halex2005 committed:
Adding WaitForDownload test for .NET

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SeleniumHQ/seleniumhq.github.io?pullRequest=1885) <br/>All committers have signed the CLA.

diemol commented:
@halex2005 can you please sign the CLA?

halex2005 commented:
@diemol, signed

harsha509 committed:
Merge branch 'trunk' into trunk

shbenzer commented:
looks like the failing check is unrelated

shbenzer commented:
looks like the failing check is unrelated @diemol

A1exKH reviewed (dismissed):
@halex2005 resolve conflicts for this PR, please.

Code LGTM.

diemol committed:
Merge branch 'trunk' into trunk

diemol committed:
Merge branch 'trunk' into trunk

diemol commented:
@halex2005, there are compilation issues. 

> cannot convert from 'string' to 'int'

diemol committed:
Merge branch 'trunk' into trunk

diemol committed:
[dotnet] Fix WaitForDownload test to use V150 DevTools API

The CDP DevToolsEventReceivedEventArgs.EventData is a JsonElement, not a
string-keyed dictionary, so the merge from trunk didn't compile. Use the
strongly-typed Browser.DownloadProgress event instead, matching the pattern
already used by the SetCookie/PerformanceMetrics tests in this file, and
correct the CSharp gh-codeblock line ranges across all locales.

diemol committed:
[dotnet] Address Qodo review: make WaitForDownload fully async and leak-proof

Replace the blocking ManualResetEvent.WaitOne with an awaited
TaskCompletionSource, and download into a unique per-run temp directory
that's cleaned up in a finally block so failed assertions don't leave
artifacts behind or risk cross-test interference.
